Server/src/actionHandlers.ts

```diff
@@ -144,75 +148,91 @@ const playHandAction = (
 			// If no lives are left, we end the game
 			if (lobby.host.lives === 0 || lobby.guest.lives === 0) {
 				const gameWinner =
-					lobby.host.lives > lobby.guest.lives ? lobby.host : lobby.guest
+					lobby.host.lives > lobby.guest.lives ? lobby.host : lobby.guest;
 				const gameLoser =
-					gameWinner.id === lobby.host.id ? lobby.guest : lobby.host
+					gameWinner.id === lobby.host.id ? lobby.guest : lobby.host;
 
-				gameWinner?.sendAction({ action: 'winGame' })
-				gameLoser?.sendAction({ action: 'loseGame' })
-				return
+				gameWinner?.sendAction({ action: "winGame" });
+				gameLoser?.sendAction({ action: "loseGame" });
+				return;
 			}
 		}
 
-		roundWinner.sendAction({ action: 'endPvP', lost: false })
-		roundLoser.sendAction({ action: 'endPvP', lost: true })
+		roundWinner.sendAction({ action: "endPvP", lost: false });
+		roundLoser.sendAction({ action: "endPvP", lost: true });
 	}
-}
+};
 
 const stopGameAction = (client: Client) => {
-	client.lobby?.broadcastAction({ action: 'stopGame' })
-}
+	client.lobby?.broadcastAction({ action: "stopGame" });
+};
 
 const gameInfoAction = (client: Client) => {
-	client.lobby?.sendGameInfo(client)
-}
-	
-const lobbyOptionsAction = (options: any, client: Client) => {
-	client.lobby?.setOptions(options)
-}
+	client.lobby?.sendGameInfo(client);
+};
+
+const lobbyOptionsAction = (
+	options: Record<string, string>,
+	client: Client,
+) => {
+	client.lobby?.setOptions(options);
+};
 
 const failRoundAction = (client: Client) => {
-	const lobby = client.lobby
+	const lobby = client.lobby;
 
-	if (!lobby) return
+	if (!lobby) return;
 
 	if (lobby.options.death_on_round_loss) {
-		client.lives -= 1
-		client.sendAction({ action: 'playerInfo', lives: client.lives })
+		client.lives -= 1;
+		client.sendAction({ action: "playerInfo", lives: client.lives });
 	}
 
 	if (client.lives === 0) {
-		let gameLoser = null
-		let gameWinner = null
+		let gameLoser = null;
+		let gameWinner = null;
 		if (client.id === lobby.host?.id) {
-			gameLoser = lobby.host
-			gameWinner = lobby.guest
+			gameLoser = lobby.host;
+			gameWinner = lobby.guest;
 		} else {
-			gameLoser = lobby.guest
-			gameWinner = lobby.host
+			gameLoser = lobby.guest;
+			gameWinner = lobby.host;
 		}
 
-		gameWinner?.sendAction({ action: 'winGame' })
-		gameLoser?.sendAction({ action: 'loseGame' })
+		gameWinner?.sendAction({ action: "winGame" });
+		gameLoser?.sendAction({ action: "loseGame" });
 	}
-}
+};
 
-const setAnteAction = ({ ante }: ActionHandlerArgs<ActionSetAnte>, client: Client) => {
-	client.ante = ante
-}
+const setAnteAction = (
+	{ ante }: ActionHandlerArgs<ActionSetAnte>,
+	client: Client,
+) => {
+	client.ante = ante;
+};
+
+/** Verifies the client version and allows connection if it matches the server's */
+const versionAction = (
+	{ version }: ActionHandlerArgs<ActionVersion>,
+	client: Client,
+) => {
+	if (version !== serverVersion) {
+		client.sendAction({ action: "error", message: "Version mismatch" });
```

Comment:
```suggestion
		client.sendAction({ action: "error", message: "WARN: Server expecting version 0.1.2" });
```

We should make this more verbose for the user

Reply:
You're right